executor/insert_common.go

```diff
@@ -513,6 +510,22 @@ func (e *InsertValues) adjustAutoIncrementDatum(d types.Datum, hasValue bool, c
 	return casted, nil
 }
 
+func getAutoRecordID(d types.Datum, target *types.FieldType) (int64, error) {
```

Comment:
this function is different from https://github.com/pingcap/tidb/pull/11110

Reply:
I'll check it.